### PR TITLE
Contracts development

### PR DESCRIPTION
**Pull request by mateigraura**

### CODE REVIEW

File: contracts/VotingCore.sol [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    constructor() public {
+        owner = msg.sender;
+    }
+
+    function setCandidates(
+        bytes32[] memory names,
+        bytes32[] memory parties,
+        bytes32[] memory imgUrls
+    ) public {
+        require(msg.sender == owner, "!owner");
```
Comment (stefanandy) [inline contracts/VotingCore.sol:35]: make a modifier 'onlyOwner'

File: contracts/VotingCore.sol [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            }));
+        }
+    }
+
+    function getCandidates() public view returns (Candidate[] memory) {
+        return candidates;
+    }
+
+    function whitelistAddress(address voterAddress) public {
+        require(msg.sender == owner, "!owner");
```
Comment (stefanandy) [inline contracts/VotingCore.sol:52]: make a modifier 'onlyOwner'

File: contracts/VotingCore.sol [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        Voter storage voter = voters[msg.sender];
+        require(voter.isAllowed, "Not allowed!");
+        require(!voter.voted, "Address already voted!");
+        voter.voted = true;
+        voter.candidateId = id;
+
+        candidates[id].count += 1;
+    }
+
+    function getWinner() public view returns (
```
Comment (stefanandy) [inline contracts/VotingCore.sol:75]: what stops someone to call getWinner function before election complete ?

**Review by stefanandy — CHANGES_REQUESTED** [review]

File: contracts/VotingCore.sol [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    uint8 data;
+
+    function setData(uint8 _data) public {
+        data = _data;
+    }
+
+    function getData() public view returns (uint8) {
+        return data;
+    }
+}
```
[the remark this answers is not in the document]
Comment (mateigraura) [reply contracts/VotingCore.sol:93]: obsolete

File: contracts/VotingCore.sol (hunk 3 above)
Comment (mateigraura) [reply contracts/VotingCore.sol:75]: for now it can stay open. it needs to be split anyway

**Review by mateigraura — COMMENTED** [review]
_

**Pull request merged**